server/src/main/java/org/elasticsearch/health/GetHealthAction.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.cluster.ClusterName;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.health.ClusterHealthStatus;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class GetHealthAction extends ActionType<GetHealthAction.Response> {
+
+    public static final GetHealthAction INSTANCE = new GetHealthAction();
+    // TODO: Need new name maybe
+    // cluster:health/get
+    public static final String NAME = "cluster:monitor/health2";
+
+    private GetHealthAction() {
+        super(NAME, GetHealthAction.Response::new);
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        private final ClusterName clusterName;
+        private final List<Component> components;
+
+        public Response(ClusterName clusterName, List<Component> components) {
+            this.clusterName = clusterName;
+            this.components = components;
+        }
+
+        public Response(StreamInput in) {
+            throw new AssertionError("GetHealthAction should not be sent over the wire.");
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            throw new AssertionError("GetHealthAction should not be sent over the wire.");
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+            builder.startObject();
```

Comment:
WDYT about adding vertical separation to make clear the boundaries between JSON objects?
```java
builder.startObject();
{
    builder.field("timed_out", false);
    builder.field("status", "green");
    builder.field("cluster_name", clusterName.value());
    builder.array("impacts");
    builder.startObject("components");
    {
        for (Component component : components) {
            builder.startObject(component.getName());
            {
                builder.field("status", component.getStatus());
                builder.startArray("indicators");
                {
                    List<Indicator> indicators = component.getIndicators();
                    for (Indicator indicator : indicators) {
                        builder.startObject();
                        {
                            builder.field("name", indicator.getName());
                            builder.field("status", indicator.getStatus());
                            builder.field("explain", indicator.getExplain());
                            builder.startObject("meta");
                            {
                                indicator.toXContent(builder, params);
                            }
                            builder.endObject();
                            // TODO: Add detail / documentation
                        }
                        builder.endObject();
                    }
                }
                builder.endArray();
            }
            builder.endObject();
        }
    }
    builder.endObject();
}
return builder.endObject();
```